hologram.cpp

```diff
@@ -1,33 +1,18 @@
+#include <cuda_runtime_api.h>
+
+#include "common/data_format.hpp"
+#include "common/phonebook.hpp"
 #include "common/plugin.hpp"
-#include "common/threadloop.hpp"
 #include "common/switchboard.hpp"
-#include "common/phonebook.hpp"
-#include "common/data_format.hpp"
-#include "common/logger.hpp"
-#include "common/threadloop.hpp"
 #include "hologram.h"
-#include <chrono>
-#include <memory>
-#include <thread>
-#include <cstdio>
-#include <cstdlib>
-#include <atomic>
-#include <cuda_runtime_api.h>
 
 using namespace ILLIXR;
-using std::unique_ptr;
-using std::thread;
-using std::atomic;
 
-class hologram : public threadloop {
+class hologram : public plugin {
```

Comment:
Looks all good. Just wondering why we want to change it to a plugin. Does this mean that we don't want it to drop any inputs? Wouldn't this cause inputs to queue up?